core/src/serve_repair.rs

```diff
@@ -664,7 +664,10 @@ mod tests {
                 .into_iter()
                 .filter_map(|b| {
                     assert_eq!(repair_response::nonce(&b.data[..]).unwrap(), nonce);
-                    Shred::new_from_serialized_shred(b.data.to_vec()).ok()
+                    use solana_ledger::shred::SHRED_PAYLOAD_SIZE;
+                    let mut serialized_shred = vec![0; SHRED_PAYLOAD_SIZE];
+                    serialized_shred.copy_from_slice(&b.data[..SHRED_PAYLOAD_SIZE]);
+                    Shred::new_from_serialized_shred(serialized_shred).ok()
```

Comment:
this block of logic appears in a bunch of places, maybe blockstore should have a common `get_padded_data` function?